File: extrusion/solutions/extrusion_diagonal.java
import java.io.*;
import java.util.*;

class Solution {
    /**
    * Creates the output image by drawing edges on each diagonal. When a corner
    * is found, keep track of how many more slashes to draw for that edge. When
    * an intersection is found with an upcoming edge, reset the number of
    * slashes.
     */
    static void solve(int H, int W, int D, char[][] base) {
        char[][] image = new char[H + D + 1][W + D + 1];
        for (int r = 0; r < H + D + 1; r++) {
            Arrays.fill(image[r], ' ');
        }
        
        for (int r = 0; r < H; r++) {
            for (int c = 0; c < W; c++) {
                image[r][c] = base[r][c];
            }
        }
        
        for (int d = -W + 1; d < H; d++) {
            int r = Math.max(0, d);
            int c = r - d;
            int slashesToDraw = 0;
            while (r < H + D + 1 && c < W + D + 1) {
                if (slashesToDraw > 0) {
                    image[r][c] = '\\';
                    slashesToDraw--;
                }
                if (r < H && c < W && base[r][c] == '+') {
                    slashesToDraw = D;
                }
                r++;
                c++;
            }
        }
        
        for (int r = 0; r < H; r++) {
            for (int c = 0; c < W; c++) {
                if (base[r][c] != ' ') {
                    image[r + D + 1][c + D + 1] = base[r][c];
                }
            }
        }
        
        for (int r = 0; r < H + D + 1; r++) {
            out.println(image[r]);
        }
    }
    
    static BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
    static PrintWriter out = new PrintWriter(System.out);

    public static void main(String[] args) throws IOException {
        int T = Integer.parseInt(in.readLine());
        for (int i = 0; i < T; i++) {
            String[] HWD = in.readLine().split(" ");
            int H = Integer.parseInt(HWD[0]);
            int W = Integer.parseInt(HWD[1]);
            int D = Integer.parseInt(HWD[2]);
            char[][] base = new char[H][];
            for (int j = 0; j < H; j++) {
                base[j] = in.readLine().toCharArray();
            }
            solve(H, W, D, base);
        }
        out.flush();
    }
}

File: extrusion/solutions/extrusion_diagonal.py
def solve(H: int, W: int, D: int, base: list[list[str]]) -> None:
    """
    Creates the output image by drawing edges on each diagonal. When a corner
    is found, keep track of how many more slashes to draw for that edge. When
    an intersection is found with an upcoming edge, reset the number of
    slashes.
    """
    image = [[' ' for _ in range(D + W + 1)] for _ in range(D + H + 1)]
    
    for i in range(H):
        for j in range(W):
            image[i][j] = base[i][j]
    
    for d in range(-W + 1, H):
        i = max(0, d)
        j = i - d
        slashesToDraw = 0;
        while i < H + D + 1 and j < W + D + 1:
            if slashesToDraw > 0:
                image[i][j] = '\\'
                slashesToDraw -= 1
            if i < H and j < W and base[i][j] == '+':
                slashesToDraw = D
            i += 1
            j += 1 
    
    for i in range(H):
        for j in range(W):
            if base[i][j] != ' ':
                image[i + D + 1][j + D + 1] = base[i][j]
    
    for i in range(H + D + 1):
        print(''.join(image[i]))

def main():
    T = int(input())
    for _ in range(T):
        H, W, D = (int(i) for i in input().split(' '))
        base = [list(input()) for _ in range(H)]
        solve(H, W, D, base)

if __name__ == '__main__':
    main()
